beluga_ros/tools/conversion_utils.py

```diff
@@ -0,0 +1,273 @@
+        self._grid: Dict[DiscreteCell, NormalDistribution] = {}
+
+    def add_distribution(self, cell: DiscreteCell, ndt: NormalDistribution):
+        """Add a new cell with its distribution."""
+        self._grid[cell] = ndt
+
+    def _get_plot_bounds(self):
+        min_x = float("inf")
+        min_y = float("inf")
+
+        max_x = -float("inf")
+        max_y = -float("inf")
+        for discrete_cell in self._grid.keys():
+            x = discrete_cell.x * self._resolution + self._resolution / 2
+            y = discrete_cell.y * self._resolution + self._resolution / 2
+            min_x = min(min_x, x)
+            max_x = max(max_x, x)
+
+            min_y = min(min_y, y)
+            max_y = max(max_y, y)
+
+        PADDING = 1  # [m]
+
+        min_x -= PADDING
+        max_x += PADDING
+        min_y -= PADDING
+        max_y += PADDING
+
+        step_x = (max_x - min_x) / 100.0
+        step_y = (max_y - min_y) / 100.0
+
+        xx, yy = np.mgrid[min_x:max_x:step_x, min_y:max_y:step_y]
+        return (xx, yy, np.dstack((xx, yy)))
+
+    def __eq__(self, other: "NDTMap") -> bool:
+        """Equality for two NDT maps."""
+        return (
+            np.allclose(self._resolution, other._resolution)
+            and self._grid == other._grid
+        )
+
+    def plot(self, show: bool = False) -> plt.figure:
+        """
+        Plot the map using contour plots into the current figure and returns it.
+
+        Optionally show the plot based on the 'show' parameter.
+        """
+        xx, yy, pos = self._get_plot_bounds()
+
+        for ndt in self._grid.values():
+            plt.contour(
+                xx,
+                yy,
+                ndt.get_scipy_representation().pdf(pos),
+                [0.1, 0.5],
+                alpha=0.2,
+                linewidths=3,
+                extend="max",
+            )
+        if show:
+            plt.show()
+        return plt.gcf()
+
+    def to_hdf5(self, output_file_path: Path):
+        """
+        Serialize the NDT map into an HDF5 format.
+
+        See https://docs.hdfgroup.org/hdf5/develop/_intro_h_d_f5.html for details on the format.
+        It'll create 4 datasets:
+            - "resolution": () resolution for the discrete grid (cells are resolution x
+              resolution m^2 squares).
+            - "discrete_cells": (NUM_CELLS, 2) that contains the discrete cell coordinates.
+            - "means": (NUM_CELLS, 2) that contains the 2d mean of the normal distribution
+              of the cell.
+            - "covariances": (NUM_CELLS, 2, 2) Contains the covariance for each cell.
+        """
+        assert output_file_path.suffix in (".hdf5", ".h5")
+        output_file_path.parent.mkdir(parents=True, exist_ok=True)
+
+        num_cells = len(self._grid.keys())
+
+        with h5py.File(output_file_path.absolute(), "w") as fp:
+            discrete_cells_dataset = fp.create_dataset(
+                "discrete_cells", (num_cells, 2), chunks=True
```

Comment:
@serraramiro1 nit^2: `discrete_cells` -> `cells`, discretization is implied.

Reply:
Done